mne/utils/misc.py

```diff
@@ -475,3 +478,35 @@ def wrapper(*args, **kwargs):
             return f(*args, **kwargs)
 
     return wrapper
+
+
+def _auto_weakref(function):
+    """Create weakrefs to self (or other free vars in __closure__) then evaluate."""
```

Comment:
```suggestion
    """Create weakrefs to self (or other free vars in __closure__) then evaluate.
    
    When a nested function is defined within an instance method, and the function makes
    use of ``self``, it creates a reference cycle that the Python garbage collector is
    not smart enough to resolve, so the parent object is never GC'd. (The reference to
    ``self`` becomes part of the ``__closure__`` of the nested function).

    This decorator allows the nested function to access ``self`` without increasing the
    reference counter on ``self``, which will prevent the memory leak. If the referent
    is not found (usually because already GC'd) it will short-circuit the decorated
    function and return ``None``.
    """
```

@larsoner I took a stab at distilling the explanation, please edit where I got things muddled

Reply:
I think it's good enough!